compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -765,6 +765,15 @@ typeOf' = \case
       throwWithContext (EWrongInterfaceRequirement requiringIface requiredIface)
     checkExpr expr (TCon requiredIface)
     pure (TOptional (TCon requiringIface))
+  EToTypeRep iface expr -> do
+    checkExpr expr (TCon iface)
```

Comment:
as far as I can tell this typechecks even if `iface` is not an interface. Should we check that and error out?